service/controller/resource/ensurecpcrs/create.go

```diff
@@ -80,6 +81,40 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 }
 
 func (r *Resource) createAWSControlPlaneCR(ctx context.Context, cr infrastructurev1alpha2.AWSCluster, id string) error {
+	cp := &infrastructurev1alpha2.AWSControlPlane{
+		TypeMeta: infrastructurev1alpha2.NewAWSControlPlaneTypeMeta(),
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				annotation.Docs: "https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2#AWSControlPlane",
+			},
+			Labels: map[string]string{
+				label.OperatorVersion: key.OperatorVersion(&cr),
+				label.Cluster:         key.ClusterID(&cr),
+				label.Organization:    key.OrganizationID(&cr),
+				label.Release:         key.ReleaseVersion(&cr),
+			},
+			Name: id,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: infrastructurev1alpha2.NewG8sControlPlaneTypeMeta().APIVersion,
+					Kind:       infrastructurev1alpha2.NewG8sControlPlaneTypeMeta().Kind,
+					Name:       id,
+				},
+			},
+		},
+		Spec: infrastructurev1alpha2.AWSControlPlaneSpec{
+			AvailabilityZones: []string{
+				key.MasterAvailabilityZone(cr),
```

Comment:
@calvix @tuommaki I need a refresher on some AZ magic. Is it possible to change the AZs here once you selected them and ran a master? Anything that can go horribly wrong under the hood?

Reply:
changing AZ means changing subnets, that's always a pain since you need to first terminate all resources in the subnet and only then you can remove the subnet, that might be hard to properly do because of race conditions
The problem is also with storage as the EBS is AZ locked, so we would also need to do some snapshot and then create new EBS in new AZ from that snapshot.

I would rather not allow change of AZs for already existing master

It's not impossible but definitely require kinda complex logic and probabbly a few steps migration

Reply:
I see what you mean. Assuming we treat the Control Plane stacks like Node Pools and launch additional ones, you can actually set different AZs for the new Control Plane stack and tear down the old one. I think this would not even cause downtime and snapshotting would also not need to happen I think.

Reply:
Well new control plane would have clear etcd so that wont work without some magic of restoring data

Reply:
I dont see any easy how how change the control plane stack

Reply:
The idea would pretty much be to let additional masters join the existing clusters. The other two masters do that already in the existing stack. So adding more masters to the cluster should kind of work the same way and it should not matter in which CF stack they are as there has to happen some sort of discovery anyway.